src/Type/NonexistentParentClassType.php

```diff
@@ -152,6 +152,11 @@ public function toArrayKey(): Type
 		return new ErrorType();
 	}
 
+	public function isOffsetAccessLegal(): TrinaryLogic
+	{
+		return TrinaryLogic::createYes();
```

Comment:
I think this should be `no`, and I think we can test it with `$p = new parent();` and trying offset access on that.

Reply:
You are definitely right! Thank you.

Also, found an interesting result that NonExistentParentClassType behavior have changed from 7.4.

We cannot even declare a NonExistentParentClassType after PHP 8.0. 
https://3v4l.org/7TR10

Calling array access is an error across all versions.
https://3v4l.org/ClM1B

Reply:
We can just skip the file from linting in Makefile https://github.com/phpstan/phpstan-src/blob/78d3d99d9306b045a5464c19c9afcca5e5d7b1e5/Makefile#L76